### ISSUE TITLE
Anime4KCPPDS slows down playing speed by mutiple times with interpolation filter like DmitriRender or SVP Pro on.

### ISSUE THREAD

**Issue by catshitz** [TianZerL/Anime4KCPP#36]
Hardware: 4K monitor+1070Ti+3600x
Software: LAV Filters+madVR+DmitriRender/SVP Pro+Anime4KCPPDS+PotPlayer/MPC-HC

I understand the interpolation filters mutiple the frames by several times yet the Anime4KCPPDS still handle it as a normal one, so what we got here is a video track that plays abnormally slow.

I also notice that Anime4KCPPDS works well with madVR, I thought they are both upscaling filters, are they essentially different?

As a matter of fact,I just want to get both realtime 4K and 60FPS effects at the same time, maybe it's too hard for this device or is there a compatibility issue between upscaling software like Anime4KCPPDS and interpolating software like DmitriRender?

BTW, is there anyway to get there regardless of whatever it takes?

**Comment by TianZerL** [comment]
If you have an AMD card, you can use bluesky frame converter.  And I recommend upscaling first, beacuse even if you have enough computing power of your video card, you need a large enough video memory bandwidth (GDDR5 is not enough, GDDR6 is needed) to process a 60fps video to 4k in realtime. So, upscaleing in before filling in frames.

**Comment by catshitz** [comment]
Thx for the quick reply,
> If you have a GCN AMD card (Intel UHD is ok but the effect may not be good enough), you can use Bluesky Frame Rate Converter
Actually I had been using BlueskyFRC for a year before when I had a RX380 card, but after I changed it to GTX and DmitriRender I found the latter one has less artifacts.
>load Anime4KCPPDS first and then load Bluesky. I recommend upscaling first
With DmitriRender/SVP Pro I found these filters must be ahead of Anime4KCPPDS, or the players will simply crash when opening a video...
>you need a large enough video memory bandwidth (GDDR5 is not enough, GDDR6 is needed)
So it is a hardware issue then, maybe a RTX 2060Super with 8GB GDDR6 video memory is needed  at least...

**Comment by TianZerL** [comment]
> So it is a hardware issue then, maybe a RTX 2060Super with 8GB GDDR6 video memory is needed at least...
I just did a quick test, GTX1660S can process 60fps 720P to 2K in 57 FPS in YV12 color format, 54FPS in NV12 color format, which have 336GB/s video memory bandwidth, so I think maybe 5700XT which have 448GB/s can get it. And I think the best one is Redeon VII with 1193GB/s bandwidth and 13.8TFlops FP32 computing power.

**Comment by catshitz** [comment]
> the best one is Redeon VII with 1193GB/s bandwidth and 13.8TFlops FP32 computing power.

That sure is a hack lot of computing power( and dollars). Anyway, thanks for your help, I'll manage to explore more about Anime4KCPP for now.

**Comment by catshitz** [comment]
Sorry, forgot to mention just now, I noticed that Anime4KCPP doesn't support mpeg-ts files with h264/avc video tracks, will ts file be supported in the future?

**Comment by TianZerL** [comment]
> Sorry, forgot to mention just now, I noticed that Anime4KCPP doesn't support mpeg-ts files with h264/avc video tracks, will ts file be supported in the future?

Is that a blu ray original file? Anime4KCPPDS support BD original file. Anime4KCPPDS don't care about file suffixes, it only care about the color format which sent form decoder, see [here](https://github.com/TianZerL/Anime4KCPP/wiki/DirectShow-filter), so you may need to check your decoder settings.

**Comment by TianZerL** [comment]
BTW, I just noticed that ffdshow need YV12 input. The output color format of Anime4KCPPDS is same as the input which is NV12 for MPC-HC/potplayer by default. So you may need to set it to YV12 in LAV decoder and then you can try to load Anime4KCPPDS first.

**Comment by catshitz** [comment]
> Is that a blu ray original file?
> the color format which sent form decoder

It's a standard YUV 4:2:0 8bit h264 m3u8 stream video, output color format NV12 in MPC-HC/PotPlayer by default.

> you may need to set it to YV12 in LAV decoder and then you can try to load Anime4KCPPDS first.

I tried, with the same YUV 4:2:0 8bit m3u8 stream video, I disabled NV12 in LAV Video Decoder both in PotPlayer and internal one in MPC-HC, left YV12 untouched which is checked by default, then I put only Anime4KCPPDS and ffdshow raw video filter in the filter area and Anime4KCPPDS ahead, the players are still crashing just like I mention before, reverse the order then it'll be fine but only slows down the playing speed.

I just tried to upscale a video cutscene for some test through Anime4KCPP, the result looks all right but frankly speaking, not as remarkble as the same output from Topaz Video Enhance AI, but the processing speed is quite promising, comparing the dozens of times it'll take in Topaz Video Enhance AI, and that's the first and exact reason I came here and check out Anime4KCPP. First I was hoping to find a good realtime upscaling method, but now I have to give up due to my low-end hardware, but hey, thanks anyway!
